### PR TITLE
[prometheus-rabbitmq-exporter] Create secret to store credentials when provided

### PR DESCRIPTION
**Pull request by jchanam**
#### What this PR does / why we need it
Connection credentials should be stored securely using secrets. Right now, the chart allows to provide your own secret, but if the credentials are given as values to the chart, they are exposed in the deployment env variables.

This PR adds a new secret, that will be created only if the user or password is provided and if the existing user/password secret is not given.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-rabbitmq-exporter/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{{- if or (and (.Values.rabbitmq.password) (not .Values.rabbitmq.existingPasswordSecret)) (and (.Values.rabbitmq.user) (not .Values.rabbitmq.existingUserSecret)) }}
```
Comment (desaintmartin) [inline charts/prometheus-rabbitmq-exporter/templates/secret.yaml:1]: If `.Values.rabbitmq.password` is set but not `.Values.rabbitmq.existingPasswordSecret`, secret will not be created and Pod will never start. We should have a solution. Maybe we should prevent the two being set at the same time, or we align the condition in the two files, or we simply "ignore" the latter like it's done before this PR.

**Review by desaintmartin — CHANGES_REQUESTED** [review]
Thanks for the contribution! This one has been on my list for a while but I never had the time to do it.

File: charts/prometheus-rabbitmq-exporter/templates/secret.yaml (hunk 1 above)
Comment (jchanam) [reply charts/prometheus-rabbitmq-exporter/templates/secret.yaml:1]: Hi!

Yes, you're right. If existing password secret is set, then we don't need to create the secret, as we'll be using the one provided in the values.

On that case, the secret where we're reading the value for the env is also the one provided, so I think the pod will be created and running.

This is an example of the env for the deployment on that case:

```yaml
          env:
            - name: RABBIT_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: "existingsecret"
                  key: "password"
            - name: RABBIT_USER
              valueFrom:
                secretKeyRef:
                  name: "existingsecret"
                  key: "username"
            - name: RABBIT_URL
              value: my-url.com
```

Comment (jchanam) [reply charts/prometheus-rabbitmq-exporter/templates/secret.yaml:1]: About the options you mentioned, I followed what it was done before. If the existing secret is given, the password or user is ignored and the secret used.

Comment (desaintmartin) [reply charts/prometheus-rabbitmq-exporter/templates/secret.yaml:1]: I am sorry, the diff bugged me, you are right! To be sure for the future, I'll add a few tests in a next PR.

**Review by desaintmartin — APPROVED** [review]

**Pull request merged**